File: ISP63_BackEnd2024/src/main/java/com/backend/ISP63/dto/UserDTO.java
package com.backend.ISP63.dto;

import com.backend.ISP63.enums.*;
import com.backend.ISP63.model.Carrera;
import com.backend.ISP63.model.Localidad;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserDTO {

    private String dni; // Usamos DNI como clave primaria
    private String apellido;
    private String nombre;
    private Genero genero;
    private String domicilio;
    private Localidad localidad;
    private String telefono;
    private String email;
    private LocalDate fechaNacimiento;
    private EstadoCivil estadoCivil;
    String username;
    private String rol;
    private CarreraDTO carrera;//  solo tendra carrera si es estudiante
}

File: ISP63_BackEnd2024/angular/src/app/modelo/users/UserDTO.ts
import { Carrera } from "../carreras/carrera";
import { EstadoCivil } from "../enums/EstadoCivil";
import { Genero } from "../enums/Genero";
import { Localidad } from "../localidad";

export interface UserDTO {
    dni: string;  // Usamos DNI como clave primaria
    apellido: string;
    nombre: string;
    genero: Genero;  // Asegúrate de tener una definición o tipo para Genero
    domicilio: string;
    localidad: Localidad;  // Asegúrate de tener una definición o tipo para Localidad
    telefono: string;
    email: string;
    fechaNacimiento: string;  // Puedes utilizar el tipo string para LocalDate
    estadoCivil: EstadoCivil;  // Asegúrate de tener una definición o tipo para EstadoCivil
    username: string;
    rol: string;
    carrera?: Carrera;  // solo tendrá carrera si es estudiante, por lo que lo hacemos opcional
}
